static/app/views/alerts/rules/metric/details/index.tsx

```diff
@@ -199,11 +201,16 @@ class MetricAlertDetails extends Component<Props, State> {
     const timePeriod = this.getTimePeriod(selectedIncident);
     const {start, end} = timePeriod;
     try {
-      const [incidents, rule] = await Promise.all([
+      const promiseList: Promise<any>[] = [
         fetchIncidentsForRule(organization.slug, ruleId, start, end),
         rulePromise,
-      ]);
+      ];
```

Comment:
inserting it into the array can avoid using Promise<any> and it should infer the types
```suggestion
			organization.features.includes('anomaly-detection-alerts') ? fetchAnomaliesForRule(organization.slug, ruleId, start, end) : undefined
      ];
```